libs/clients/assets-v2/src/lib/assetsV2.config.ts

```diff
@@ -0,0 +1,53 @@
+import { NationalRegistryScope } from '@island.is/auth/scopes'
+import { defineConfig } from '@island.is/nest/config'
+import { z } from 'zod'
+
+const schema = z.object({
+  xRoadServicePath: z.string(),
+  fetch: z.object({
+    timeout: z.number().int(),
+    autoAuth: z
+      .object({
+        mode: z.enum(['token', 'tokenExchange', 'auto']),
+        issuer: z.string(),
+        clientId: z.string(),
+        clientSecret: z.string(),
+        scope: z.array(z.string()),
+      })
+      .optional(),
+  }),
+})
+
+export const AssetsV2ClientConfig = defineConfig<z.infer<typeof schema>>({
+  name: 'AssetsClient',
+  schema,
+  load(env) {
+    const clientSecret = env.optional('XROAD_PROPERTIES_CLIENT_SECRET')
+    return {
+      xRoadServicePath: env.required(
+        'XROAD_PROPERTIES_SERVICE_V2_PATH',
+        'IS-DEV/GOV/10033/HMS-Protected/Fasteignir-v1',
+      ),
+      fetch: {
+        timeout: env.optionalJSON('XROAD_PROPERTIES_TIMEOUT') ?? 15000,
+        autoAuth: clientSecret
+          ? {
+              mode: 'tokenExchange',
+              issuer: env.required(
+                'IDENTITY_SERVER_ISSUER_URL',
+                'https://identity-server.dev01.devland.is',
+              ),
+              clientId:
+                env.optional('XROAD_PROPERTIES_CLIENT_ID') ??
+                '@island.is/clients/national-registry',
+              clientSecret,
+              scope: env.optionalJSON('XROAD_PROPERTIES_SCOPE') ?? [
+                NationalRegistryScope.properties,
+                'api_resource.scope',
+              ],
+            }
+          : undefined,
```

Comment:
You should reference the `charge-fjs-v2` client for the auto auth setup.

In short:
Instead of defining autoAuth here you should do it in your `PropertiesV2ApiProvider` and use the `IdsClientConfig` from `@island.is/nest/config`. This simplifies your config schema as it only needs to contain the `autoAuth.scope` config and optionally the `fetch.timeout`.

The v1 `assets` client is using the "old" version where we initially were going to start with machine client per organisation integration. But it should be using the `api` machine client for token exchange, as we are going forward with one machine client per service, and the `IdsClientConfig` is a utility to minimise the boilerplate.

And as always feel free to ping me on slack to if anything is unclear :)